agent/grpc-external/services/peerstream/stream_test.go

```diff
@@ -1219,36 +1226,52 @@ func expectReplEvents(t *testing.T, client *MockClient, checkFns ...func(t *test
 }
 
 func TestHandleUpdateService(t *testing.T) {
-	srv, _ := newTestServer(t, func(c *Config) {
+	srv, store := newTestServer(t, func(c *Config) {
 		backend := c.Backend.(*testStreamBackend)
 		backend.leader = func() bool {
 			return false
 		}
 	})
 
 	type testCase struct {
-		name   string
-		seed   []*structs.RegisterRequest
-		input  *pbservice.IndexedCheckServiceNodes
-		expect map[string]structs.CheckServiceNodes
+		name                          string
+		seed                          []*structs.RegisterRequest
+		input                         *pbservice.IndexedCheckServiceNodes
+		expect                        map[string]structs.CheckServiceNodes
+		expectedImportedServicesCount int
 	}
 
 	peerName := "billing"
+	peerID := "1fabcd52-1d46-49b0-b1d8-71559aee47f5"
 	remoteMeta := pbcommon.NewEnterpriseMetaFromStructs(*structs.DefaultEnterpriseMetaInPartition("billing-ap"))
 
 	// "api" service is imported from the billing-ap partition, corresponding to the billing peer.
 	// Locally it is stored to the default partition.
 	defaultMeta := *acl.DefaultEnterpriseMeta()
 	apiSN := structs.NewServiceName("api", &defaultMeta)
 
+	// create a peering in the state store
+	require.NoError(t, store.PeeringWrite(31, &pbpeering.Peering{
+		ID:   peerID,
+		Name: peerName},
+	))
+
+	// connect the stream
+	_, err := srv.Tracker.Connected(peerID)
+	require.NoError(t, err)
+
 	run := func(t *testing.T, tc testCase) {
+
+		mst, found := srv.Tracker.MutableStreamStatus(peerID)
```

Comment:
This is fine but you wouldn't have needed for this to be in the stream tracker. For the purposes of checking for imported service counts it would have been fine to just create a new MutableStatus outside of the tracker and pass it in.